### PR TITLE
Gomplate tests / changes

### PR DESCRIPTION
**Pull request by drekle**
Small refactor of self render

### CODE REVIEW

**Comment by dcwangmit01** [comment]
Are there any functionality changes?

File: mhlib/app.go [hunk 1]
```diff
@@ -294,64 +294,70 @@ func (a *App) render(configFile string) (*string, *string, *[]byte, error) {
...
+	type gomplateConfig struct {
+		Gomplate gomplate.Config `yaml:"gomplate,omitempty"`
+	}
+	var gomp gomplateConfig
+	err := yaml.Unmarshal([]byte(templateValuesStr), &gomp)
+	if err != nil {
+		return "", err
+	}
+
+	funcs := template.FuncMap{
```
Comment (dcwangmit01) [inline mhlib/app.go]: Are you adding a function?  Perhaps we shuld upstream this function into gomplate instead?  The guy is really responsive.

File: mhlib/app_test.go [hunk 2]
```diff
@@ -1,67 +1,68 @@
...
-		t.Log(err)
-		t.Fail()
-	}
-	if out != expected {
-		t.Logf("Actual: %s\nExpected: %s\n", out, expected)
-		t.Fail()
+	if out, err := selfRender(templateString); out != expected {
+		if err != nil {
+			t.Log(err)
+			t.FailNow()
```
Comment (dcwangmit01) [inline mhlib/app_test.go]: Why not run through all of the tests, and then fail later?

Comment (drekle) [reply mhlib/app_test.go]: This would be easier to debug a failed test.

**Comment by drekle** [comment]
There are Functional changes.  I am unmarshalling the gomplate at the gomplate tag.  this was previously trying to unmarshal DataSources and Datasource Headers at root.

File: mhlib/app.go (hunk 1 above)
Comment (drekle) [reply mhlib/app.go]: I am not sure I am using this function anymore.  I could remove.  This should make struct elements intensive however this is no longer a struct,  I think it probably should be, however the structure of our yaml does not support this.

**Pull request merged**